wagtail/admin/views/generic/base.py

```diff
@@ -199,6 +214,96 @@ def filter_queryset(self, queryset):
             queryset = self.filters.filter_queryset(queryset)
         return queryset
 
+    def get_url_without_filter_param(self, param):
+        """
+        Return the index URL with the given filter parameter removed from the query string
+        """
+        base_url = self.index_url.split("?")[0]
+        query_dict = self.request.GET.copy()
+        query_dict.pop(self.page_kwarg, None)  # reset pagination to first page
+        if isinstance(param, (list, tuple)):
+            for p in param:
+                query_dict.pop(p, None)
+        else:
+            query_dict.pop(param, None)
+        return base_url + "?" + query_dict.urlencode()
+
+    def get_url_without_filter_param_value(self, param, value):
+        """
+        Return the index URL where the filter parameter with the given value has been removed
+        from the query string, preserving all other values for that parameter
+        """
+        base_url = self.index_url.split("?")[0]
+        query_dict = self.request.GET.copy()
+        query_dict.pop(self.page_kwarg, None)  # reset pagination to first page
+        query_dict.setlist(
+            param, [v for v in query_dict.getlist(param) if v != str(value)]
+        )
+        return base_url + "?" + query_dict.urlencode()
```

Comment:
Drive-by question: would the `set_query_param` util be of any use here?

https://github.com/wagtail/wagtail/blob/0ee97d45deacae6e210207e3ebdd3b0b94afb4c7/wagtail/admin/utils.py#L59-L72

Reply:
It'll do the job for `get_url_without_filter_param` (but not `get_url_without_filter_param_value`), but I'd rather avoid the overhead of parsing a URL when we already have a QueryDict.